src/app/core/entity-details/related-entities-with-summary/related-entities-with-summary.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { NgFor, NgIf } from "@angular/common";
+import { DynamicComponent } from "../../config/dynamic-components/dynamic-component.decorator";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { EntitySubrecordComponent } from "../../common-components/entity-subrecord/entity-subrecord/entity-subrecord.component";
+import { RelatedEntitiesComponent } from "../related-entities/related-entities.component";
+import { Entity } from "../../entity/model/entity";
+import { filter } from "rxjs/operators";
+import { applyUpdate } from "../../entity/model/entity-update";
+
+/**
+ * Load and display a list of entity subrecords (entities related to the current entity details view)
+ * including a summary below the table.
+ */
+@DynamicComponent("RelatedEntitiesWithSummary")
+@UntilDestroy()
+@Component({
+  selector: "app-related-entities-with-summary",
+  templateUrl: "./related-entities-with-summary.component.html",
+  imports: [EntitySubrecordComponent, NgIf, NgFor],
+  standalone: true,
+})
+export class RelatedEntitiesWithSummaryComponent<E extends Entity = Entity>
+  extends RelatedEntitiesComponent<E>
+  implements OnInit
+{
+  /**
+   * Configuration of what numbers should be summarized below the table.
+   */
+  @Input() summaries?: {
+    countProperty: string;
+    groupBy?: string;
+    total?: boolean;
+    average?: boolean;
+  };
+
+  summary = "";
+  avgSummary = "";
+
+  async ngOnInit() {
+    await super.ngOnInit();
+    this.updateSummary();
+
+    this.entityMapper
+      .receiveUpdates(this.entityCtr)
+      .pipe(
+        untilDestroyed(this),
+        filter(
+          ({ entity, type }) =>
+            type === "remove" || entity[this.property] === this.entity.getId(),
+        ),
+      )
+      .subscribe((update) => {
+        this.data = applyUpdate(this.data, update);
+        this.updateSummary();
+      });
+  }
+
+  /**
+   * update the summary or generate a new one.
+   * The summary contains no duplicates and is in a
+   * human-readable format
+   */
+  updateSummary() {
+    if (!this.summaries) {
+      this.summary = "";
+      this.avgSummary = "";
+      return;
+    }
+
+    const summary = new Map<string, { count: number; sum: number }>();
+    const average = new Map<string, number>();
+
+    this.data.forEach((m) => {
+      const amount = m[this.summaries.countProperty];
+      let groupLabel;
+      if (this.summaries.groupBy) {
+        groupLabel =
+          m[this.summaries.groupBy]?.label ?? m[this.summaries.groupBy];
+      }
+
+      summary.set(groupLabel, summary.get(groupLabel) || { count: 0, sum: 0 });
```

Comment:
good point!

I now added code to ensure this simply displays the sum/average without any label, in case there is only undefined group (which is also the case if the now optional "groupBy" was not given).

If there is groupings, that will be displayed as something like "groupA: 5, undefined: 3", which is useful in my opinion.